util.c

```diff
@@ -227,6 +227,28 @@ check_firmware_version(YK_KEY *yk, bool verbose, bool quiet)
 }
 
 int
+check_user_group(char *username, char *group)
+{
+  struct group *group_ptr;
+  char **group_member;
+  
+  if((group_ptr = getgrnam(group)) == NULL) {
```

Comment:
I'm not certain about whether or not getgrnam_r provides any additional checking over getgrnam, as they both essentially return a list of members of a specified group. But I believe the biggest difference between the two is that getgrnam stores the data in a single buffer and returns a pointer to it, but the buffer is overwritten on subsequent calls. However, I don't mind making the modifications to use getgrnam_r instead as it would provide a starting grounds for expansion with regards to remote authentication and threading... I have some ideas for developing a means of making challenge-response capable of remotely authenticating via a centralized server and using SSL to keep the challenge/response safe between the client/server. 

We can chat more about that outside of here some time. If you could drop me an email at jhall@futuresouth.us, I'd love to send you over the details of what I'd like to take on and contribute to the project.